RecoTracker/MkFitCore/standalone/Makefile.config

```diff
@@ -130,6 +130,7 @@ endif
 ifeq ($(CXX), g++)
   CXXFLAGS += -std=c++1z -ftree-vectorize -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -Wstrict-overflow -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wunused -Wparentheses -Wno-deprecated -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Wstrict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi
   CXXFLAGS += -fdiagnostics-color=auto -fdiagnostics-show-option -pthread -pipe -fopenmp-simd
+  CXXFLAGS += -Ofast -fno-reciprocal-math -mrecip=none
```

Comment:
is `-fno-math-errno` still needed?

Reply:
I'm not 100% sure. IIUC, @srlantz tested the additional flags on top of the existing ones, that's why I did not removed it. However, can be removed if unnecessary.

Reply:
also , `-ftree-vectorize` is a part of `-O2`, apparently (which is in turn a part of `-O3`, and that is in `-Ofast`)

or is the point that redundant flags are OK?

Reply:
I let @srlantz comment about it: to me, we can clean the unnecessary flags; however, I thought I would be as close as possible to what I understood was tested as a start.

Reply:
In general, not setting errno is a requirement for vectorizing transcendentals, so it’s probably worth specifying even if it’s subsumed by other optimization flags

Reply:
To be entirely clear about it, yes, this PR is good with me   :-)

Reply:
it may help to add a few comments about the redundant flags (either that it helps to ensure vectorization regardless of being redundant or that it was copied from CMSSW [add a cite/ref])

Reply:
@slava77, do you mean in the PR description (in progress), or directly in the Makefile?

Reply:
If the update of the description is enough, I will open a PR to cms-sw/cmssw

Reply:
I mean the Makefile itself

Reply:
I agree with @slava77 that it's good to add a few comments to the Makefile. Prior to the big chunk of CXXFLAGS, maybe put this:
```# For gcc, compile with flags approximating "scram build echo_CXXFLAGS" in a CMSSW env```
...because here's what I get when I do that command using `cmsrel CMSSW_12_6_X_2023-04-14-1100`:
```CXXFLAGS = -O2 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++1z -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -msse3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-deprecated-copy -Wno-unused-parameter -Wunused -Wparentheses -Wno-deprecated -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi```
The first few are different, but the others concerning the warnings/errors are largely the same? Maybe the differences arise because the env has shifted over time?

Reply:
Then ahead of the new flags, I would maybe say:
```
# Add flags to enable auto-vectorization of sin/cos with libmvec (glibc >= 2.22)
```

Reply:
One can dump the big lines of CXXFLAGS from `scram build echo_CXXFLAGS` and Makefile.config into two files (flags1, flags2) and git-diff them to see what the exact differences are:
```
$ git diff --word-diff --no-index -- flags1 flags2
diff --git a/flags1 b/flags2
index b730cec4ab6..7a7e1926904 100644
--- a/flags1
+++ b/flags2
@@ -1 +1 @@
CXXFLAGS [-= -O2 -pthread -pipe-]{++= -std=c++1z -ftree-vectorize+} -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow [--std=c++1z -ftree-vectorize-]{+-Wstrict-overflow+} -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections[--msse3-] -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type[--Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-deprecated-copy -Wno-unused-parameter-] -Wunused -Wparentheses -Wno-deprecated -Werror=return-type -Werror=missing-braces -Werror=unused-value[--Werror=unused-label-] -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor [--Werror=strict-aliasing-]{+-Wstrict-aliasing+} -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi
```
Basically flags1 (from scram) has about a dozen more flags than flags2 (from our Makefile.config). A few others are merely moved around or changed slightly.

Reply:
To make things easier (I hope), I introduced a commit to insert the comments I suggested. But I still wonder if we've got the right CXXUSERFLAGS to reproduce what happens with the CMSSW build of mkFit, given the differences noted above. Should we edit the flags to match more closely the output of `scram build echo_CXXFLAGS`, based on a recent CMSSW version?

Reply:
Thanks, @srlantz. I have been traveling and attending the remote CMS week, and had just no time to come back to this sooner. Apologies.
From 13_1_0_pre3, here's the list of flags returned by `scram build echo_CXXFLAGS`:

`CXXFLAGS = -O2 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++17 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -fuse-ld=bfd -msse3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-deprecated-copy -Wno-unused-parameter -Wunused -Wparentheses -Wno-deprecated -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi`

This is the diff wrt. the mkFit `Makefile.config`:
`[--O2 -pthread -pipe-]{+-std=c++1z -ftree-vectorize+} -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow [--std=c++17 -ftree-vectorize-]{+-Wstrict-overflow+} -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections[--fuse-ld=bfd -msse3-] -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type[--Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-deprecated-copy -Wno-unused-parameter-] -Wunused -Wparentheses -Wno-deprecated -Werror=return-type -Werror=missing-braces -Werror=unused-value[--Werror=unused-label-] -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor [--Werror=strict-aliasing-]{+-Wstrict-aliasing+} -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi`

Reply:
I'm not sure whether we should edit `Makefile.config` to be closer to the CMSSW default flags. But if there's no reason not to, we could go with it?

Reply:
The Makefile.config gcc flags were copied from CMSSW at some point in the distant past, it would be completely reasonable to update to whatever the defaults are today.  (I think there was one flag I had to omit back in the day, but that must have been resolved getting mkFit into CMSSW.)

Reply:
Looks good to me. I agree with @dan131riley that we should take the opportunity to update these flags. Thanks @mmasciov !

Reply:
Thanks! Then, if OK with everyone, I will open a PR to cms-sw/cmssw